src/data-loaders/RelationalDataLoader.js

```diff
@@ -0,0 +1,179 @@
+  return rows.map((row) => {
+    const typeName = Object.keys(pgTypes.builtins)
+      .find((d) => pgTypes.builtins[d] === row.dataTypeID)
+      .toUpperCase();
+    // @TODO: Add support for the following fields
+    // isAutoIncrement,
+    // nullable,
+    // isSigned,
+    // autoIncrement,
+    // tableName,
+    // arrayBaseColumnType,
+    // isCaseSensitive,
+    // isCurrency,
+    // currency,
+    // precision,
+    // scale,
+    // schemaName,
+    return {
+      label: row.name,
+      name: row.name,
+      type: row.dataTypeID,
+      typeName,
+    };
+  });
+};
+const executeSqlStatements = async (client, req) =>
+  Promise.mapSeries(req.statements, (statement) => client.query(statement));
+export default class RelationalDataLoader {
+  constructor(config) {
+    this.config = config;
+  }
+
+  async load(req) {
+    try {
+      const requiredKeys = [
+        'dbDialect',
+        'dbUsername',
+        'dbPassword',
+        'dbHost',
+        'dbName',
+        'dbPort',
+      ];
+      if (!this.config.rds) {
+        throw new Error('RDS configuration not passed');
+      }
+      const missingKey = requiredKeys.find((key) => {
+        return !this.config.rds[key];
+      });
+      if (missingKey) {
+        throw new Error(`${missingKey} is required.`);
+      }
+
+      const dbConfig = {
+        host: this.config.rds.dbHost,
+        user: this.config.rds.dbUsername,
+        password: this.config.rds.dbPassword,
+        database: this.config.rds.dbName,
+        port: this.config.rds.dbPort,
+      };
+
+      const res = {};
+      if (this.config.rds.dbDialect === 'mysql') {
+        const client = await mysql.createConnection(dbConfig);
+        const results = await executeSqlStatements(client, req);
+
+        res.sqlStatementResults = results.map((result) => {
+          if (result.length < 2) {
+            return {};
+          }
+          if (!result[1]) {
+            // not a select query
+            return {
+              numberOfRecordsUpdated: result[0].affectedRows,
+              generatedFields: [],
+            };
+          }
+          return {
+            numberOfRecordsUpdated: result[0].length,
+            records: convertSQLResponseToRDSRecords(result[0]),
+            columnMetadata: convertMySQLResponseToColumnMetaData(result[1]),
+          };
+        });
+      } else if (this.config.rds.dbDialect === 'postgres') {
+        const client = new Client(dbConfig);
+        await client.connect();
+        const results = await executeSqlStatements(client, req);
+        res.sqlStatementResults = results.map((result) => {
+          return {
+            numberOfRecordsUpdated: result.rowCount,
+            records: convertSQLResponseToRDSRecords(result.rows),
+            columnMetadata: convertPostgresSQLResponseToColumnMetaData(
+              result.fields,
+            ),
+            generatedFields: [],
+          };
+        });
+      }
+      return JSON.stringify(res);
```

Comment:
Hey @alichherawalla Thanks for this 👍 

But I wonder, why are you returning with a `json string`, instead of a `json object` here? 

I'm testing this on our project, and stumbled into this 🤔

Reply:
Hi, 

That's the format that we get it in the VTL. 

Happy to help if you're running into issues with your use case.

Reply:
Not a problem, in our case, it conflicts with the deployed expected type (on AWS) and offline local. So I had to do something like this

```
#set ($result = $utils.rds.toJsonObject($ctx.result))
#if (!$result)
    #set ($result = $utils.rds.toJsonString($ctx.result))
#end
```

Reply:
Ah, I actually use this for both the AWS deployed version and local development and it works fine. And I'm using it for both MySQL and PostgreSQL.

Please correct me if you think otherwise or find some resources that better explain it but according to the documentation here: https://docs.aws.amazon.com/appsync/latest/devguide/tutorial-rds-resolvers.html

> In the response mapping template section, add the following template:
> $utils.toJson($utils.rds.toJsonObject($ctx.result)[1][0])

and the documentation of `utils.rds.toJsonObject` states that it accepts a serializedSQLResult

>$util.rds.toJsonObject(String serializedSQLResult): Object
> Same as util.rds.toJsonString but with the result to be a JSON Object.

Based on the above, `$ctx.result` must be a string too. Also `toJsonString` actually accepts a `serializedSQLResult` as per the documentation.

>$util.rds.toJsonString(String serializedSQLResult): String
> Returns a String by transforming the stringified raw RDS Data API result format to a more concise string.

Reply:
Yup, we used to do it like this:
```
#set ($result = $utils.rds.toJsonObject($ctx.result)[0])
#if ($result.isEmpty()) #return #else $utils.toJson($result) #end
```
This is exactly what you propose... but the result in the first bit is always `null` because we are receiving a String instead of an Object.

Now I have to do something like this, to catch when DataLoader returns a String representation of an object
```
#set ($result = $utils.rds.toJsonObject($ctx.result)[0])
#if (!$result)
    #set ($result = $utils.rds.toJsonString($ctx.result))
    #if ($result.isEmpty()) #return #else $util.toJson($util.parseJson($result)[0]) #end
#else
    #if ($result.isEmpty()) #return #else $util.toJson($result)) #end
#end
```

And it just works 😅

Reply:
Hey @katesclau!
Is there any way you know to avoid having to do this check?
The solution you mentioned that 'just works' does indeed work, but isn't there something wrong with the `RelationalDataLoader` implementation if we need to do these special checks? 
Shouldn't it behave 1:1 with AWS? I tried to remove the JSON.stringify'ing, but it still kept returning null.

I have a project with about 257 response templates and I'm not looking that forward to having to replace them all 😬

Reply:
Hi @jariz @katesclau,

IMO this isn't related to the implementation of the Relational Dataloader, it's because the [amplify-appsync-simulator](https://github.com/aws-amplify/amplify-cli/blob/master/packages/amplify-appsync-simulator/src/velocity/util/rds.ts) doesn't implement the `toJsonObject` as yet.

I will raise a PR with amplify and add support for it.

Reply:
Thanks a ton 😁

Reply:
Decided to open a PR myself :)

aws-amplify/amplify-cli#8021